packages/amplify_core/lib/src/config/analytics/pinpoint_config.dart

```diff
@@ -119,3 +128,13 @@ class PinpointTargeting with AWSEquatable<PinpointTargeting>, AWSSerializable {
   @override
   Map<String, Object?> toJson() => _$PinpointTargetingToJson(this);
 }
+
+class DurationConverter implements JsonConverter<Duration, int> {
+  const DurationConverter();
+
+  @override
+  Duration fromJson(int json) => Duration(seconds: json);
+
+  @override
+  int toJson(Duration object) => object.inSeconds;
```

Comment:
This will round down to the nearest whole second. Is that what we want? If so, can we document that on `autoFlushEventsInterval`? If a customer created their amplify config using `AmplifyConfig` class, they might be surprised to see to see it rounded.

Reply:
Might be just better to accept an `int` in the constructor and have Duration be the getter. If you do that, though, then I would rename the field in the ctor as `autoFlushEventsIntervalSecs` and then you can remove this converter class altogether